### PR TITLE
refactor (jkube-kit/enricher) : Move storageClass related functionality out of VolumePermissionEnricher (#1179)

### LINKED ISSUE #1179: Avoid initContainer while mounting a volume
### Description
Since I switch from fabric8 to JKUBE, When I add a volume in my description.yaml.
An initContainer using busybox is automatically added, was not the case with fabric8.

The main issue, my openshift instance dont have access to busybox image (due to corporate restriction)
### Info

* Eclipse JKube version : 1.4.0
* Maven version (`mvn -v`) : 3.8.1
```

```
* Kubernetes / Red Hat OpenShift setup and version : 3.11

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Closes #1179
    
+ Make BaseEnricher's `getConfig` method public in order to make an enricher access other enricher's config
+ Move setting PersistentVolumeClaim's storageClass related logic into a
     new enricher : PersistentVolumeClaimStorageClassEnricher . This way user can
     simply exclude VolumePermissionEnricher to opt out of volume permission
     initContainer.
+ Deprecate VolumePermissionEnricher's `defaultStorageClass` and
      `useStorageClassAnnotation` fields in favor of
      PersistentVolumeClaimStorageClassEnricher equivalents.

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [X] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [X] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ protected Configuration getConfiguration() {
         return enricherContext.getConfiguration();
     }
 
-    protected String getConfig(Configs.Config key) {
+    public String getConfig(Configs.Config key) {
```
Comment (manusa) [inline jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java:105]: why is the change of access needed?

Comment (rohanKanojia) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java:105]: I did it to access VolumePermissionEnricher's Config fields from PersistentVolumeClaimStorageClassEnricher .

Comment (manusa) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java:105]: OK, that reveals a big code-smell.
In any case (now that I'm seeing the affected part) if you wanted to expose that method you could have done that at the VolumePermissionEnricher level (by overriding the method and making it public).

Comment (manusa) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java:105]: So,for now, the recommended changes would be to remove this change of access here, and change it at the VolumePermissionEnricher level with a `// TODO` comment linking an issue where we indicate that we need to remove that once the deprecated configs are finally removed in a next version.

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**